### PR TITLE
[COST-3247] Properly handle manifest files from Azure

### PR DESCRIPTION
**Pull request by samdoran**
## Jira Ticket

[COST-3247](https://issues.redhat.com/browse/COST-3247)

## Description

Properly handle manifest files in Azure.

## Testing

1. Checkout Branch
```
git fetch upstream pull/4064/head:PR/4064
git switch PR/4064
```
2. Restart Koku
3. Add an Azure source using a partitioned export
4. Verify that data is properly processed.

## Notes

Once this is merged, the language in the docs stating _not_ to enable partitioning can be removed.

### CODE REVIEW

File: koku/masu/external/downloader/azure/azure_report_downloader.py [hunk 1]
```diff
@@ -141,12 +143,50 @@ def _get_manifest(self, date_time):
...
+                cost_export_blob = self._azure_client.get_latest_cost_export_for_path(report_path, self.container_name)
+            except AzureCostReportNotFound as ex:
+                msg = f"Unable to find cost export. Error: {ex}"
+                LOG.info(log_json(self.tracing_id, msg, self.context))
+                return manifest, None
+
+            report_name = cost_export_blob.name
+            last_modified = cost_export_blob.last_modified
+            LOG.info(log_json(self.tracing_id, f"Found cost export {report_name}", self.context))
+            manifest["reportKeys"] = [report_name]
```
Comment (maskarb) [inline koku/masu/external/downloader/azure/azure_report_downloader.py:189]: You COULD reduce complexity here by moving some of this out to smaller methods. Then you should be able to removed the C901:
```suggestion
        if json_manifest:
            try:
                manifest_json, last_modified = self.get_json_manifest(json_manifest)
            except AzureReportDownloaderError as err:
                msg = f"Unable to get report manifest for {self._provider_uuid}. Reason: {str(err)}"
                LOG.info(log_json(self.tracing_id, msg, self.context))
                return None, None
            manifest["reportKeys"] = [blob["blobName"] for blob in manifest_json["blobs"]]
        else:
            try:
                report_name, last_modified = self.get_cost_export(report_path)
            except AzureCostReportNotFound as ex:
                msg = f"Unable to find cost export. Error: {ex}"
                LOG.info(log_json(self.tracing_id, msg, self.context))
                return None, None
            manifest["reportKeys"] = [report_name]
```

and then add:
```

    def get_cost_export(self, report_path):
        cost_export_blob = self._azure_client.get_latest_cost_export_for_path(report_path, self.container_name)
        report_name = cost_export_blob.name
        last_modified = cost_export_blob.last_modified
        LOG.info(log_json(self.tracing_id, f"Found cost export {report_name}", self.context))
        return report_name, last_modified

    def get_json_manifest(self, json_manifest):
        report_name = json_manifest.name
        last_modified = json_manifest.last_modified
        LOG.info(log_json(self.tracing_id, f"Found JSON manifest {report_name}", self.context))

        # Download the manifest and extract the list of files.
        manifest_tmp = self._azure_client.download_file(
            report_name, self.container_name, suffix=AzureBlobExtension.json.value
        )
        # Extract data from the JSON file
        try:
            with open(manifest_tmp) as f:
                manifest_json = json.load(f)
        except json.JSONDecodeError as err:
            msg = f"Unable to open JSON manifest. Reason: {err}"
            raise AzureReportDownloaderError(msg)
        finally:
            self._remove_manifest_file(manifest_tmp)
        return manifest_json, last_modified
```

**Review by maskarb — COMMENTED** [review]
Things look fine to me except for that one return statement. And one other suggestion, but it's not a necessary change.

**Review by maskarb — APPROVED** [review]
I ingested from `SA1` and `partitionedcostexport` and the data looks consistent to me! LGTM!

File: koku/masu/external/downloader/azure/azure_report_downloader.py (hunk 1 above)
Comment (samdoran) [reply koku/masu/external/downloader/azure/azure_report_downloader.py]: I like this approach. I was pretty spent on Friday and wasn't up for more refactoring. I'll work on this in a follow up PR.

**Pull request merged**